manticore/core/smtlib/constraints.py

```diff
@@ -165,13 +167,14 @@ def declarations(self):
         for a in self.constraints:
             try:
                 declarations.visit(a)
-            except BaseException:
-                # there recursion limit exceeded problem,
-                # try a slower, iterative solution
-                #logger.info('WARNING: using iterpickle to dump recursive expression')
-                #from utils import iterpickle
-                #file('recursive.pkl', 'w').write(iterpickle.dumps(a))
-                raise
+            except RuntimeError:
+                # TODO: (defunct) move recursion management out of PickleSerializer
+                if sys.getrecursionlimit() >= PickleSerializer.MAX_RECURSION:
```

Comment:
Since these constants are being used for smtlib visitors and pickle serialization, should we move them outside of `PickleSerializer`? This might tie with the global config PR.

Reply:
:100: , global config seems like the place for them to live. i'll open an issue to move them once global config is implemented.

re;#1058